### PR TITLE
cmd/acme: be more careful with pipes in X commands

### LINKED ISSUE #9: cmd/acme: Edit X/.../ broken
Acme's implementation of the X sam looping construct crashes with error:

  qunlock pc=0x1079c6b30 owner=0 self=108124000 oops

How to reproduce?
1. Execute 'New foo'.
2. Execute 'Edit X/foo/ >wc'.

### PR DESCRIPTION
**Pull request by jacobvosmaer**
This is an alternative to https://github.com/9fans/plan9port/pull/222 . The goal is to fix #9 and #219: Acme crashes when an `X` or `Y` edit visits a window different from where the command was written, and if the loop body invokes a pipe (`<`, `>` or `|`).

### CODE REVIEW

File: src/cmd/acme/ecmd.c [hunk 1]
```diff
@@ -626,8 +628,12 @@ runpipe(Text *t, int cmd, Rune *cr, int ncr, int state)
...
 	run(w, runetobyte(s, n), dir.r, dir.nr, TRUE, nil, nil, TRUE);
 	free(s);
-	if(t!=nil && t->w!=nil)
-		winunlock(t->w);
+	/*
+	 * Only winunlock if we are in the window the command originated from.
+	 * With X and Y this is not automatically the case.
+	 */
+	if(t!=nil && t->w!=nil && (Xorigwin==nil || Xorigwin==t->w))
+			winunlock(t->w);
```
Comment (jacobvosmaer) [inline src/cmd/acme/ecmd.c]: Oops one tab too many

**Comment by rsc** [comment]
I don't think this is right either. I think the lock really does need to be released around the pipe execution.

**Comment by jacobvosmaer** [comment]
I don't really understand what I'm doing but I wonder if the behavior of this PR is unclear.

@rsc as I understand it this change keeps the same `winunlock` behavior we had before, with the exception that during X (`Xorigwin!=nil`), we iterate over windows some of which weren't locked to begin with. So while `Xorigwin!=nil` we should _only_ unlock if the the window the current text is in is the "originating window" (`Xorigwin==t->w`).

Sorry if all that was obvious.

Are you saying we should `winlock` all windows that X visits before we run commands on them?

**Comment by jacobvosmaer** [comment]
@rsc is 1ad3ad52 then what you had in mind?

**Pull request closed without merging**